source/Halibut/Transport/Streams/WriteIntoMemoryBufferStream.cs

```diff
@@ -94,7 +99,12 @@ public override int WriteTimeout
         public override void SetLength(long value) => throw new NotSupportedException();
         public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
 
-        public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        protected override Task<int> _ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return innerStream.ReadAsync(buffer, offset, count, cancellationToken);
```

Comment:
Do we actually read on this stream? Or should it do what `Read` does, and `throw new NotSupportedException();`?

Reply:
Good point done